### ISSUE TITLE
Unable to find documentation around "t.has" pattern

### ISSUE THREAD

**Issue by MoSheikh** [tapjs/tapjs#763]
Hi, 

Thanks for the great project! I was wondering if you had any documentation around how the "t.has()" tests should be formed. I tried looking around but wasn't able to find an example of what the "pattern" argument mentioned in the docs should look like in practice.

Any help would be greatly appreciated.

**Comment by mandaputtra** [comment]
I also confuse on how to do this, with has

**Comment by isaacs** [comment]
Doc patch welcome, or I'll update when I get some time. Basically, it's the same as t.match(), but any specified fields in the pattern have to match exactly. (Another way to say it is that it's like `t.same()`, but any extra fields in the object are ignored.)

This passes: `t.has({a:1,b:2}, {a:1})` but this doesn't: `t.has({a:1}, {a:1,b:2})`.

**Comment by isaacs** [comment]
Docs: https://node-tap.org/docs/api/asserts/#thasfound-pattern-message-extra

**Comment by mandaputtra** [comment]
I'll patch it, I thought `t.has` doesn't care about the value it provided

```
  t.has({ a: 1 }, { a: 2 });
```

Will fail, where do I find something like this `t.hasProperty({ a: 1, b: 2}, ['a'])` so it only checks if the property is there and doesn't care about the value. Is there any?

**Comment by isaacs** [comment]
The easiest way to test for a values existence/type but not value is using `t.match()`.

For example:

```js
t.match({ a: 1 }, { a: Number }) // passes
t.match({ a: 1 }, { a: String }) // fails
```

It's a little bit weird/clever but you could also use `notHas` for this, ironically.

```js
t.notHas({ a: 1 }, { a: undefined }) // passes, because the a it has is not undefined
```

But that's very counterintuitive to read, so I wouldn't recommend it.  I think it'd be good to have a `t.hasProperties(obj, nameList)` that verifies all the named properties are not undefined, and `t.hasOwnProperties(obj, list)` that verifies that it has own properties by all the names in the list.  What do you think?

**Comment by isaacs** [comment]
Haha, decided to not use `hasOwnProperty` for the assertion, because overwriting an object's `hasOwnProperty` is just a footgun waiting to happen.  Feedback welcome on tapjs/libtap#44

**Comment by isaacs** [comment]
Published in 15.2.0 https://node-tap.org/docs/api/asserts/#thaspropfound-property-message-extra
